src/v/config/node_config.cc

```diff
@@ -18,102 +18,107 @@ node_config::node_config() noexcept
     *this,
     "data_directory",
     "Place where redpanda will keep the data",
-    required::yes)
+    {.required = required::yes, .visibility = visibility::user})
   , node_id(
       *this,
       "node_id",
       "Unique id identifying a node in the cluster",
-      required::yes)
-  , rack(*this, "rack", "Rack identifier", required::no, std::nullopt)
+      {.required = required::yes, .visibility = visibility::user})
+  , rack(
+      *this,
+      "rack",
+      "Rack identifier",
+      {.visibility = visibility::user},
+      std::nullopt)
   , seed_servers(
       *this,
       "seed_servers",
       "List of the seed servers used to join current cluster. If the "
       "seed_server list is empty the node will be a cluster root and it will "
       "form a new cluster",
-      required::no,
+      {.visibility = visibility::user},
```

Comment:
`visibility::user`  is a default visibility, why do we need to specify it here?

Reply:
my guess is that being explicit makes it a bit more readable. but I'm curious as well.

`struct metadata` might benefit from one or more constructors, or builder interface like `metadata().set_user().set_secret()` but I dunno if its worth it.

Reply:
I'm using an explicit visibility setting to encourage folks to think about it when adding/editing properties (realistically, when people create a property they're doing it with copy-paste from an existing one).

We could clean this up further by using constructors to require that parameter while making others optional, but that makes everything a lot more verbose in the context of long lists of property definitions.

Reply:
+1